library/init/category/lift.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2016 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner, Sebastian Ullrich
+-/
+prelude
+import init.function init.coe
+import init.category.monad
+import init.util
+
+universes u v w
+
+/-- A function for lifting a computation from an inner monad to an outer monad.
+    Like https://hackage.haskell.org/package/transformers-0.5.5.0/docs/Control-Monad-Trans-Class.html,
+    but `n` does not have to be a monad transformer. -/
+class has_monad_lift (m : Type u → Type v) (n : Type u → Type w) :=
+(monad_lift {} : ∀ {α}, m α → n α)
+
+/-- The reflexive-transitive closure of `has_monad_lift`. -/
+class has_monad_lift_t (m : Type u → Type v) (n : Type u → Type w) :=
+(monad_lift {} : ∀ {α}, m α → n α)
+
+export has_monad_lift_t (monad_lift)
+
+@[reducible] def has_monad_lift_to_has_coe {m n} [has_monad_lift_t m n] {α} : has_coe (m α) (n α) :=
```

Comment:
Does this generic coercion work reliably? 
We have tried similar coercions in the past, and they worked very poorly.
The problem is the coercion resolution procedure. 
Issue #1402 contains more info.

Reply:
This is not a new definition; Github gets confused by the file renaming. The definition is used in super.